src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -7943,5 +7940,41 @@ public void ParamsArray_Symbol_MultipleParamsArrays()
             Assert.False(((SourceParameterSymbol)lambdas[1].Parameters[1]).IsParams);
             Assert.True(((SourceParameterSymbol)lambdas[1].Parameters[2]).IsParams);
         }
+
+        [Fact]
+        public void ParamsArray_NotLast()
+        {
+            var source = """
+                var lam = (params int[] xs, int y) => xs.Length + y;
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (1,12): error CS0231: A params parameter must be the last parameter in a parameter list
+                // var lam = (params int[] xs, int y) => xs.Length + y;
+                Diagnostic(ErrorCode.ERR_ParamsLast, "params int[] xs").WithLocation(1, 12));
+        }
+
+        [Fact]
+        public void ParamsArray_Multiple()
+        {
+            var source = """
+                var lam = (params int[] xs, params int[] ys) => xs.Length + ys.Length;
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (1,12): error CS0231: A params parameter must be the last parameter in a parameter list
+                // var lam = (params int[] xs, params int[] ys) => xs.Length + ys.Length;
+                Diagnostic(ErrorCode.ERR_ParamsLast, "params int[] xs").WithLocation(1, 12));
+        }
+
+        [Fact]
+        public void ParamsArray_NotArray()
+        {
+            var source = """
+                var lam = (params int x) => x;
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (1,12): error CS0225: The params parameter must be a single dimensional array
+                // var lam = (params int x) => x;
+                Diagnostic(ErrorCode.ERR_ParamsMustBeArray, "params").WithLocation(1, 12));
+        }
```

Comment:
Consider adding tests to cover manual usage of `ParamArrayAttribute`.

Reply:
Thanks, added: e1eac8963dff7ab516a2ceab567078006e87da70